lib/game.rb

```diff
@@ -0,0 +1,87 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+def current_player
+  #if the turn count is even, then it's going to be player 1's turn
+  if self.board.turn_count.even?
+    player_1
+  else
+    player_2
+
+  end
+  #else is player 2's
+end
+
+def won?
+  WIN_COMBINATIONS.find do |win_combo|
+    #win_combo will be an array
+    index_1 = win_combo[0]
+    index_2 = win_combo[1]
+    index_3 = win_combo[2]
+    position_1 = self.board.cells[index_1]
+    position_2 = self.board.cells[index_2]
+    position_3 = self.board.cells[index_3]
+    position_1 == position_2 && position_3 == position_1 && position_2 != " "
+
+    end
+  end
+
+  def draw?
+    self.board.full? && !won?
+  end
+
+  def over?
+    self.won? || self.draw?
+  end
+
+  def winner
+    if self.won?
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.